ambari-server/src/main/java/org/apache/ambari/server/topology/AmbariContext.java

```diff
@@ -355,7 +355,7 @@ public void createAmbariServiceAndComponentResources(ClusterTopology topology, S
     }
 
     Set<ServiceGroupRequest> serviceGroupRequests = serviceGroups.stream()
-      .map(serviceGroupName -> new ServiceGroupRequest(clusterName, serviceGroupName))
+      .map(serviceGroupName -> new ServiceGroupRequest(clusterName, serviceGroupName, Iterables.getFirst(topology.getBlueprint().getStackIds(), null).getStackId()))
```

Comment:
Iterables.getFirst() seems dangerous.  If you're relying on the first of a collection then the design is wrong.  We should only use these for unit testing.

Reply:
@ncole This is only a stopgap solution, we'll need to revise both service group creation and usage in this method in a followup change.

Reply:
@adoroszlai do we have a follow up JIRA for this?